### ISSUE TITLE
MiniScript/GreyScript doesn't populate `outer` of anonymous functions

### ISSUE THREAD

**Issue by Olipro** [ayecue/greybel-vs#115]
This is pretty absurd to me that MiniScript does this, but there we go. Behaviour is the same in GreyScript, too.

```Lua
a = function(x)
  return function
    print x
  end function
end function

b = function(x)
  someFunc function //-- calling imaginary function someFunc with anonymous callback
    print x
  end function
end function
```

Should fail with a runtime error because `x` is not in scope.

This, however, should work just fine:

```Lua
a = function(x)
  ret =function
    print x
  end function
  return @ret
end function

b = function(x)
  foo = function
    print x
  end function
  someFunc @foo //-- calling imaginary function someFunc with named callback
end function
```

**Comment by ayecue** [comment]
Hey thanks that is a really good find. Interesting behaviour from MS. Anyway I now pass a flag for assign outer and it's only true if it's within an assignment.

Should be supported when https://github.com/ayecue/greybel-vs/pull/116 is merged.

**Comment by ayecue** [comment]
Yeah it updated to 1.9.3 - https://github.com/ayecue/greybel-vs/pull/113. It included some fixes for the other issues which are open right now.

I am currently testing around with 1.9.4 since it also includes a change of the data structure of internal maps to make it faster. Additionally 1.9.4 is waiting for another small performance improvement which is about batching iterations.

**Comment by ayecue** [comment]
1.9.4 - https://github.com/ayecue/greybel-vs/pull/116 is merged now. So the fix should be available now.

**Comment by Olipro** [comment]
All good!
